src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -215,13 +209,14 @@ private async ValueTask LoadOrReloadProjectsAsync(ImmutableSegmentedList<Project
         }
         finally
         {
-            projectBuildManager.EndBatchBuild();
-
             _logger.LogInformation($"Completed (re)load of all projects in {stopwatch.Elapsed}");
+
+            if (inProcessBuildHost != null)
+                await inProcessBuildHost.ShutdownAsync();
```

Comment:
why do we need to shut down the in process build host?  Is it not re-usable across projects?

Reply:
Just kind of wondering if we could have one determination at the beginning of oop vs. inproc, then use the build host in the same way every where else.

Reply:
We do use it across projects, but we shut it down at the end so that way we're not holding onto all the memory from the builds. This is same as it was before, just with different method names now. And me reusing the BuildHost like this for the in-proc wasn't really my original intent, but it avoided duplication when adding back in the in-proc flag.